src/vision.cs

```diff
@@ -2302,5 +2607,815 @@ interface VNRequestProgressProviding {
+
+		[Export ("minimumLatencyFrameCount")]
+		nint MinimumLatencyFrameCount { get; }
+
+		[Export ("frameAnalysisSpacing")]
+		CMTime FrameAnalysisSpacing { get; }
+
+		// We must inline the following 5 properties
+		// ('Revision', 'WeakSupportedRevisions', 'SupportedRevisions', 'DefaultRevision' and 'CurrentRevision')
+		// into subclasses so the correct class_ptr is used for the static members and the right enum type is also used.
+
+		[Export ("revision")]
+		VNStatefulRequestRevision Revision { get; set; }
+
+		[Static]
+		[Export ("supportedRevisions", ArgumentSemantic.Copy)]
+		NSIndexSet WeakSupportedRevisions { get; }
+
+		[Static]
+		[Wrap ("GetSupportedVersions<VNStatefulRequestRevision> (WeakSupportedRevisions)")]
+		VNStatefulRequestRevision [] SupportedRevisions { get; }
+
+		[Static]
+		[Export ("defaultRevision")]
+		VNStatefulRequestRevision DefaultRevision { get; }
+
+		[Static]
+		[Export ("currentRevision")]
+		VNStatefulRequestRevision CurrentRevision { get; }
+	}
+
+	[Deprecated (PlatformName.MacOSX, 11, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[Deprecated (PlatformName.iOS, 14, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[Deprecated (PlatformName.TvOS, 14, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[TV (11,0), Mac (10,13), iOS (11,0)]
+	[Internal]
+	[Static]
+	interface VNVideoProcessingOptionKeys {
+		[Field ("VNVideoProcessingOptionFrameCadence")]
+		NSString FrameCadenceKey { get; }
+
+		[Field ("VNVideoProcessingOptionTimeInterval")]
+		NSString TimeIntervalKey { get; }
+	}
+
+	[Deprecated (PlatformName.MacOSX, 11, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[Deprecated (PlatformName.iOS, 14, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[Deprecated (PlatformName.TvOS, 14, 0, message: "Use 'VNVideoProcessorRequestProcessingOptions' instead.")]
+	[TV (11,0), Mac (10,13), iOS (11,0)]
+	[StrongDictionary ("VNVideoProcessingOptionKeys")]
+	interface VNVideoProcessingOptions {
+		nint FrameCadence { get; set; }
+		double TimeInterval { get; set; }
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface VNVideoProcessor {
+
+		[Export ("initWithURL:")]
+		IntPtr Constructor (NSUrl videoUrl);
+
+		[Export ("addRequest:processingOptions:error:")]
+		bool AddRequest (VNRequest request, VNVideoProcessorRequestProcessingOptions processingOptions, [NullAllowed] out NSError error);
+
+		[Export ("removeRequest:error:")]
+		bool RemoveRequest (VNRequest request, [NullAllowed] out NSError error);
+
+		[Export ("analyzeTimeRange:error:")]
+		bool Analyze (CMTimeRange timeRange, [NullAllowed] out NSError error);
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface VNVideoProcessorCadence : NSCopying {
```

Comment:
looks like a candidate for `[DisableDefaultCtor]`
even more since next type is a subclass that does have it